src/Talis/Persona/Client/Users.php

```diff
@@ -0,0 +1,248 @@
+    {
+        if(!is_string($guid) || trim($guid) === '')
+        {
+            throw new \InvalidArgumentException('Invalid guid');
+        }
+        if(!is_array($profile) || empty($profile))
+        {
+            throw new \InvalidArgumentException('Invalid profile');
+        }
+        if(!is_string($token) || trim($token) === '')
+        {
+            throw new \InvalidArgumentException('Invalid token');
+        }
+
+        $url = $this->config['persona_host'].'/users/'.$guid.'/profile';
+
+        try
+        {
+            $user = $this->personaPatchUser($url, $profile, $token);
+            return $user;
+        } catch(\Exception $e)
+        {
+            throw new \Exception('User not updated');
+        }
+    }
+
+    /* Protected functions */
+
+    /**
+     * Get a persona user
+     * @param string $url
+     * @param string $token
+     * @access protected
+     * @return mixed
+     * @throws \Exception
+     */
+    protected function personaGetUser($url, $token)
+    {
+        $curlOptions = array(
+            CURLOPT_URL             => $url,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_FOLLOWLOCATION  => true,
+            CURLOPT_TIMEOUT         => 30,
+            CURLOPT_HTTPHEADER      => array('Authorization: Bearer ' . $token)
+        );
+
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        if (isset($headers['http_code']) && $headers['http_code'] === 200)
+        {
+            return json_decode($response,true);
+        } else
+        {
+            throw new \Exception("Could not retrieve OAuth response code");
+        }
+    }
+
+    /**
+     * Create a new user in Persona
+     * @param string $url
+     * @param array $query
+     * @param string $token
+     * @throws \Exception
+     * @return array
+     */
+    protected function personaPostUser($url, $query, $token)
+    {
+        $curlOptions = array(
+            CURLOPT_POST            => true,
+            CURLOPT_URL             => $url,
+            CURLOPT_FOLLOWLOCATION  => true,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_TIMEOUT         => 30,
+            CURLOPT_POSTFIELDS      => json_encode($query),
+            CURLOPT_HTTPHEADER      => array('Authorization: Bearer ' . $token)
+        );
+
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        if (isset($headers['http_code']) && $headers['http_code'] === 200)
+        {
+            return json_decode($response,true);
+        } else
+        {
+            throw new \Exception((isset($responseDecoded['error_description']) ? $responseDecoded['error_description'] : 'Could not retrieve OAuth response code'));
```

Comment:
$respnseDecoded is missing...it was in the code that is showing as deleted in the diff above